src/Tgstation.Server.Host/Program.cs

```diff
@@ -67,6 +70,34 @@ public static async Task<int> Main(string[] args)
 				args = listArgs.ToArray();
 			}
 
+			if (InteropServices.RuntimeInformation.IsOSPlatform(InteropServices.OSPlatform.Linux))
+			{
+				using var proc = new Process
+				{
+					StartInfo = new ProcessStartInfo
+					{
+						FileName = "id",
+						Arguments = "-u",
+						UseShellExecute = false,
+						RedirectStandardOutput = true,
+						CreateNoWindow = true,
+					},
+				};
+
+				proc.Start();
+				await proc.WaitForExitAsync();
+				if (proc.ExitCode is not 0 || !int.TryParse(await proc.StandardOutput.ReadToEndAsync(), out var uid))
+				{
+					Console.Error.WriteLine("Failed to obtain user id.");
+					return 1;
+				}
+
+				if (uid is 0)
+				{
+					Console.Error.WriteLine("TGS is being run as root. This is not recommended and may prevent launch in a future version.");
+				}
```

Comment:
Fyi you can do `Environment.UserName` and just see if thats `root` or not.

Up to you if you want to check on `uid==0`, or prevent the need to spawn extra processes.

Reply:
Not a good solution as you can change the name of root